ClassCastException in Spring Web when both a controller method and an advice method return ResponseEntity

When a Spring controller is something like:

```
 @GetMapping(value = "/{id}")
    public ResponseEntity<Post> getPost(@PathVariable("id") String id) {

        Post post = this.postRepository.findById(id).orElseThrow(
                () -> new PostNotFoundException(id)
        );

        return ok(post);
    }
```

and the controller advice looks like:

```
@ExceptionHandler(PostNotFoundException.class)
    public ResponseEntity notFound(PostNotFoundException ex) {
        Map<String, String> errors = new HashMap<>();
        errors.put("entity", "POST");
        errors.put("id", "" + ex.getSlug());
        errors.put("code", "not_found");
        errors.put("message", ex.getMessage());

        return status(HttpStatus.NOT_FOUND).body(errors);
    }
```

then a ClassCastException is thrown when the controller method is used with an "illegal" ID.
Something like:

```
Caused by: java.lang.ClassCastException: class java.util.HashMap cannot be cast to class org.springframework.http.ResponseEntity
```

Originally reported [here](https://github.com/quarkusio/quarkus/issues/4042#issuecomment-552443998).

--- a/bom/runtime/pom.xml
+++ b/bom/runtime/pom.xml
@@ -16,7 +16,7 @@
     <properties>
         <aesh.version>1.11</aesh.version>
         <jandex.version>2.1.1.Final</jandex.version>
-        <resteasy.version>4.4.0.Final</resteasy.version>
+        <resteasy.version>4.4.1.Final</resteasy.version>
         <opentracing.version>0.31.0</opentracing.version>
         <opentracing-jaxrs.version>0.4.1</opentracing-jaxrs.version>
         <opentracing-web-servlet-filter.version>0.2.3</opentracing-web-servlet-filter.version>

